### PR TITLE
Building QR Code Scanner User Authentication System in Computer Vision

### LINKED ISSUE #6267: [Machine learning]Build QR Code Scanner User Authentication System in Computer Vision
### Proposal Submission
Building A QR Code Scanner  User Authentication System 
#### Proposed title of article
Build QR Code Scanner User Authentication System in Computer Vision

#### Proposed article introduction
Computer Vision has opened up a plethora of possibilities and one of the simplest and most popular day-to-day applications is the use of QR Codes and Bar Codes Scanners.
Quick Response Code (QR Code) being a matrix type code is popularly used to keep track of items and store data as pixels which correspond to binary codes that can be interpreted to reveal the actual data. With this data can be shared in encrypted form without having to reveal the encoded data explicitly. This acts as an added layer of protection to prevent the misuse of sensitive information.
In this article we will cover as much to equip the reader with the knowledge of the QR Codes, how to scan the codes from an image and how to decode the Codes in Real-Time. With this knowledge of QR codes we will build a user authentication system using the OpenCV library in computer vision that will  checks if a QR Code belongs to authorised personnel and decides whether to grant them access or not

#### Key takeaways 
The reader will be able to:
- Get know to know what QR Codes are and there application in real life.
- Understand how to scan the Code from an Image using OpenCV library.
- Understand how to Decoding QR Codes in Real-Time using camera feed input.
- Understand the implementation of QR Codes in building a user authentication system.
#### Article quality
This article will cover as much not only explaining the concept of QR Codes but also how one can scan and decode the codes. It will also cover QR Code applications in real life and there implementation in building a User Authentication System in Computer Vision.

#### References
N/A

### Templates to use as guides
N/A

### PR DESCRIPTION
**Pull request by dentonya**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6267

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[Building QR Code Scanner User Authentication System in Computer Vision #6765](https://github.com/section-engineering-education/engineering-education/files/8160306/New.Scan.9_01.AM.Copyleaks.Report.2.pdf)

**Comment by dentonya** [comment]
@Bayler done with the changes concerning the revisions

**Comment by ahmadmardeni1** [comment]
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.
Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.
Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[[Building QR Code Scanner User Authentication System in Computer Vision](https://github.com/section-engineeringeducation/engineering-education/pull/6765#)
#6765](https://github.com/section-engineering-education/engineering-education/files/8167662/New.Scan.7_40.AM.Copyleaks.Report.pdf)

**Comment by dentonya** [comment]
hello @Bayler @ahmadmardeni1 most of the revisions needed are from the code snippets of which the syntax  of the python codes and all the libraries cannot be changed. Can the review process continue?

**Comment by dentonya** [comment]
@Bayler @ahmadmardeni1  done with the changes

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

**Comment by Bayler** [comment]
@dentonya Please perform a Grammarly check on your PR.

**Comment by dentonya** [comment]
> @dentonya Please perform a Grammarly check on your PR.

ook

**Comment by dentonya** [comment]
> @dentonya Please perform a Grammarly check on your PR.

@Bayler  done with  grammar changes

File: content/articles/building-qr-code-authentication-system/index.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+from pyzbar.pyzbar import decode
+import cv2
+```
+To read the QR Code image we will use the cv2 library as shown below.
+
+```python
+image = cv2.imread("qrcodes/qr_code_txt.png")
+coded_infor = decode(image)
+print(coded_infor)
+```
```
Comment (Bayler) [inline content/articles/building-qr-code-authentication-system/index.md:67]: You need to add this image you're talking about inside your tutorial.

File: content/articles/building-qr-code-authentication-system/index.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+Quick Response codes consist of various regions that have specific importance.
+- Finder Pattern: This pattern is found in the symbol's upper left, upper right, and lower left corners. They are easily detected in all directions and ease the location and identification of the QR Code size, position, and inclination.
+- Alignment Pattern: It comprises a single central module, light (3x3), and dark (5x5) modules, and acts as a correction pattern that corrects the distorted QR Code.
+- Timing Pattern: These patterns are arranged in both horizontal and vertical positions to identify the central coordinate of each data cell and correct the central coordinate at a time of distortion.
+- Quiet Zone: This is a region free of all markings. This margin space is needed for the reading of the QR Code, while the data will be stored in the data area of the symbol.
+- Encoding Region (Cell): Data, parity modules, and decoding information are all stored in this area.
+![QR Code Structure](/engineering-education/building-qr-code-authentication-system/qr-structure.jpg)
```
Comment (Bayler) [inline content/articles/building-qr-code-authentication-system/index.md:44]: Is this image yours? If not, have you added the source?

File: content/articles/building-qr-code-authentication-system/index.md [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+### Real-Time QR Code Decoding
+In real life, QR code  scanners require users to decode the codes in real-time. To do this, we will start by importing all the libraries as described above. Then we will enable the webcam camera for the input and create a secondary output window using the code below.
+```python
+#decoding in real-time
+#scanning QR code from camera feed
+video = cv2.VideoCapture(0)
+video.set(3,640) #creating  output window
+video.set(4,740)
+```
+To find out if they can be granted access or not, we will check if the decoded text is present in the text file. If there is a match, a green-colored bounding box will be displayed with the text "Access Granted." If there is no match, a red-colored bounding box will be displayed with the message "Denied Access".
```
[the remark this answers is not in the document]
Comment (dentonya) [reply content/articles/building-qr-code-authentication-system/index.md:113]: yes it explains the code below it

**Comment by dentonya** [comment]
@Bayler done with the requested changes

**Comment by Bayler** [comment]
@dentonya You need to pay attention to all the commits made by peer reviewers on your PRs. Some mistakes you're still making yet you've been with our community for a while now. It's unacceptable. Especially mistakes with grammar checks and formatting. As I mentioned in a PR I reviewed earlier, please take your time and write a polished PR. No need to rush if you getting all these corrections. 
This PR will now proceed for final review.

**Comment by dentonya** [comment]
> @dentonya You need to pay attention to all the commits made by peer reviewers on your PRs. Some mistakes you're still making yet you've been with our community for a while now. It's unacceptable. Especially mistakes with grammar checks and formatting. As I mentioned in a PR I reviewed earlier, please take your time and write a polished PR. No need to rush if you getting all these corrections. 
> This PR will now proceed for final review.

Have always followed the instructions...but thanks for the correction..Noted @Bayler ...

**Comment by hectorkambow** [comment]
> @dentonya You need to pay attention to all the commits made by peer reviewers on your PRs. Some mistakes you're still making yet you've been with our community for a while now. It's unacceptable. Especially mistakes with grammar checks and formatting. As I mentioned in a PR I reviewed earlier, please take your time and write a polished PR. No need to rush if you getting all these corrections. This PR will now proceed for final review.

Totally agree @Bayler 

cc @dentonya

**Comment by hectorkambow** [comment]
jumping on this one

**Pull request merged**